samza-api/src/main/java/org/apache/samza/table/ReadableTable.java

```diff
@@ -58,4 +60,12 @@
    */
   void close();
 
+  /**
+   * Initializes the table before any messages are processed.
+   *
+   * @param config the {@link Config} for the application
+   * @param context the {@link TaskContext} for this task
+   */
+  default void init(Config config, TaskContext context) {
```

Comment:
Difference is that taskContext has per-task metrics registry where container context only has the global metrics registry. Since table is per-task, it's probably more meaningful to register metrics under task. 

Does this make sense?

Reply:
No, we will need to support two flavors of tables: per container and per task. We are now only implement the per task flavor. Therefore, let's make sure we don't limit ourself when we add support for per container tables.